src/bestcrypt_ve_fmt_plug.c

```diff
@@ -364,11 +390,11 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 
 		/* we now compute sha256(decrypted_content[0:0x40]) and 
 		compare it with decrypted_content[0x40:0x60] */
-		memcpy(intermediate_keys, out, 0x40);
 		SHA256_Init(&ctx);
-		SHA256_Update(&ctx, intermediate_keys, 0x40);
+		SHA256_Update(&ctx, out, 0x40);
 		SHA256_Final(sha256_hash, &ctx);
 		cracked[index] = (0 == memcmp(sha256_hash, out + 0x40, 0x20));
+		any_cracked |= 1;
```

Comment:
Please add this before the `any_cracked |= 1;` line:

```c
#ifdef _OPENMP
#pragma omp atomic
#endif
```

(The only point in using the `|=` instead of simple assignment is that the former is supported by that pragma directive. If we dare not to use the pragma anyway, then we'd go for simple assignment and hope that it is in fact atomic.)